tests/aws/services/apigateway/test_apigateway_lambda_cfn.py

```diff
@@ -1,80 +1,304 @@
+import base64
+import json
+import time
+import uuid
+
 import aws_cdk as cdk
 import aws_cdk.aws_apigateway as apigateway
+import aws_cdk.aws_iam as iam
 import aws_cdk.aws_lambda as awslambda
 import pytest
 
 from localstack.testing.pytest import markers
+from localstack.utils.strings import short_uid
+from localstack.utils.sync import retry
+from localstack.utils.testutil import get_lambda_log_events
+
+API_DESTINATION_AUTHS = [
+    {
+        "type": "BASIC",
+        "key": "BasicAuthParameters",
+        "parameters": {"Username": "user", "Password": "pass"},
+    },
+    {
+        "type": "API_KEY",
+        "key": "ApiKeyAuthParameters",
+        "parameters": {"ApiKeyName": "Api", "ApiKeyValue": "apikey_secret"},
+    },
+    {
+        "type": "OAUTH_CLIENT_CREDENTIALS",
+        "key": "OAuthParameters",
+        "parameters": {
+            # AuthorizationEndpoint will be replaced dynamically below
+            "AuthorizationEndpoint": "replace_this",
+            "ClientParameters": {"ClientID": "id", "ClientSecret": "password"},
+            "HttpMethod": "POST",
+            "OAuthHttpParameters": {
+                "BodyParameters": [
+                    {"Key": "grant_type", "Value": "client_credentials"},
+                    {"Key": "client_id", "Value": "id"},
+                    {"Key": "client_secret", "Value": "password"},
+                    {"Key": "oauthbody", "Value": "value1"},
+                ],
+                "HeaderParameters": [
+                    {"Key": "Content-Type", "Value": "application/x-www-form-urlencoded"},
+                    {"Key": "oauthheader", "Value": "value2"},
+                ],
+                "QueryStringParameters": [{"Key": "oauthquery", "Value": "value3"}],
+            },
+        },
+    },
+]
+
+LAMBDA_BACKEND_CODE = r"""
+import json
+def handler(event, context):
+    print(json.dumps(event))
+    return {
+        "statusCode": 200,
+        "headers": event.get("headers", {}),
+        "body": json.dumps(event.get("body", {})),
+        "queryStringParameters": event.get("queryStringParameters", {})
+    }
+"""
 
-FN_CODE = """
+OAUTH_TOKEN_LAMBDA_CODE = r"""
 import json
 def handler(event, context):
+    print(json.dumps(event))
     return {
         "statusCode": 200,
+        "headers": {"Content-Type": "application/json"},
         "body": json.dumps({
-            "message": "Hello World!"
+            "access_token": "my-oauth-token",
+            "token_type": "bearer",
+            "expires_in": 86400
         })
     }
 """
 
 
 @markers.acceptance_test
-class TestApigatewayLambdaIntegration:
-    @pytest.fixture(scope="class", autouse=True)
+@pytest.mark.parametrize("auth", API_DESTINATION_AUTHS)
+class TestApiDestinationsCfnComplex:
```

Comment:
question: why are we parametrizing the whole test class here when we don't use the `auth` pytest parameter in the setup fixture? We are now setting up the whole stack multiple times. 

I think the philosophy of this kind of stack is to deploy once and test multiple times, and this test goes a bit against that. We could only parametrize the `test_api_destination_complex_integration` test and keep the same stack deploy, if the test properly cleans up on assertion.